src/modal/modal.scss

```diff
@@ -11,15 +11,53 @@
   left: 0;
   width: 100%;
   height: 100%;
+  @include themed {
+    background-color: rgba(0, 0, 0, t(iui-opacity-4));
+  }
+
+  // #region Small Modal Animations for CSS
   visibility: hidden;
   opacity: 0;
   @media (prefers-reduced-motion: no-preference) {
-    transition: visibility 0s linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
+    transition: visibility $iui-speed-instant linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
   }
-  @include themed {
-    background-color: rgba(0, 0, 0, t(iui-opacity-4));
+
+  &.iui-modal-visible {
+    visibility: visible;
+    opacity: 1;
+    // remove delay for entry animation so it's instantly visible
+    transition-delay: $iui-speed-instant;
+  }
+  // #endregion Small Modal Animations for CSS
+
+  // #region Small Modal Animations for React
+  &.iui-modal-animation-enter {
+    visibility: hidden;
+    opacity: 0;
+  }
+
+  &.iui-modal-animation-enter-active {
+    visibility: visible;
+    opacity: 1;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: visibility $iui-speed-instant linear, opacity $iui-speed-fast ease-out;
+    }
+  }
+
+  &.iui-modal-animation-exit {
+    visibility: visible;
+    opacity: 1;
   }
 
+  &.iui-modal-animation-exit-active {
+    visibility: hidden;
+    opacity: 0;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: visibility $iui-speed-instant linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
+    }
+  }
+  // #endregion Full Page Modal Animations for React
```

Comment:
Is this region name correct?

Reply:
Yes, the name is correct. This is the end of the section that deals with animations for the full page modal using CSSTransitions in React. It was placed a bit off though.